app/org/sagebionetworks/bridge/services/email/BasicEmailProvider.java

```diff
@@ -0,0 +1,106 @@
+package org.sagebionetworks.bridge.services.email;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeBodyPart;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.models.studies.EmailTemplate;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+public class BasicEmailProvider implements MimeTypeEmailProvider {
+
+    private final Study study;
+    private final Set<String> recipientEmails;
+    private final Map<String,String> tokenMap;
+    private final EmailTemplate template;
+    
+    private BasicEmailProvider(Study study, Map<String,String> tokenMap, Set<String> recipientEmails, EmailTemplate template) {
+        this.study = study;
+        this.recipientEmails = recipientEmails;
+        this.tokenMap = tokenMap;
+        this.template = template;
+    }
+    public Study getStudy() {
+        return study;
+    }
+    public Set<String> getRecipientEmails() {
+        return recipientEmails;
+    }
+    public Map<String,String> getTokenMap() {
+        return ImmutableMap.copyOf(tokenMap);
+    }
+    public EmailTemplate getTemplate() {
+        return template;
+    }
+    
+    @Override
+    public MimeTypeEmail getMimeTypeEmail() throws MessagingException {
+        tokenMap.put("studyName", study.getName());
+        tokenMap.put("studyId", study.getIdentifier());
+        tokenMap.put("supportEmail", study.getSupportEmail());
+        tokenMap.put("technicalEmail", study.getTechnicalEmail());
+        tokenMap.put("sponsorName", study.getSponsorName());
+        tokenMap.put("host", BridgeConfigFactory.getConfig().getHostnameWithPostfix("webservices"));
+        
+        final MimeTypeEmailBuilder builder = new MimeTypeEmailBuilder();
+
+        final String formattedSubject = BridgeUtils.resolveTemplate(template.getSubject(), tokenMap);
+        builder.withSubject(formattedSubject);
+
+        Set<String> senderEmails = BridgeUtils.commaListToOrderedSet(study.getSupportEmail());
```

Comment:
We have a recurring problem where sometimes the support email doesn't verify their email in SES. Now that we're using this email for email verification and password reset, do you think that might be a problem?